cos-fleetshard-sync/src/main/java/org/bf2/cos/fleetshard/sync/resources/AddonCleanup.java

```diff
@@ -0,0 +1,88 @@
+package org.bf2.cos.fleetshard.sync.resources;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.bf2.cos.fleetshard.api.ManagedConnector;
+import org.bf2.cos.fleetshard.support.metrics.MetricsRecorder;
+import org.bf2.cos.fleetshard.support.resources.NamespacedName;
+import org.bf2.cos.fleetshard.support.resources.Resources;
+import org.bf2.cos.fleetshard.sync.FleetShardSyncConfig;
+import org.bf2.cos.fleetshard.sync.FleetShardSyncScheduler;
+import org.bf2.cos.fleetshard.sync.client.FleetShardClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.Temporal;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+@ApplicationScoped
+public class AddonCleanup {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AddonCleanup.class);
+
+    public static final String JOB_ID = "cos.connectors.addon.cleanup";
+    public static final String METRICS = "connectors.addon.cleanup";
+
+    private static final String OPERATORS_API = "operators.coreos.com/v1alpha1";
+    private static final String OPERATOR_KIND = "ClusterServiceVersion";
+
+    @Inject
+    KubernetesClient kubernetesClient;
+    @Inject
+    FleetShardSyncConfig config;
+    @Inject
+    FleetShardSyncScheduler scheduler;
+    @Inject
+    MeterRegistry registry;
+
+    private volatile MetricsRecorder recorder;
+
+    public void start() throws Exception {
+        LOGGER.info("Starting cleanup");
+
+        recorder = MetricsRecorder.of(registry, config.metrics().baseName() + "." + METRICS);
+
+        scheduler.schedule(
+            JOB_ID,
+            ConnectorStatusSyncJob.class,
+            config.resources().cleanupRetryInterval());
+    }
+
+    public void stop() {
+        scheduler.shutdownQuietly(JOB_ID);
+    }
+
+    public void run() {
+        recorder.record(this::cleanup);
+    }
+
+    public void cleanup() {
+        String clusterId = config.cluster().id();
+        LOGGER.info("Deleting all ManagedConnector resources that belong to cluster {}", clusterId);
+        kubernetesClient.resources(ManagedConnector.class)
```

Comment:
we may not need this one as the connectors are automatically deleted by deleting the ns.